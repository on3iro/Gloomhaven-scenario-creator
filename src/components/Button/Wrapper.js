/**
 * Styled container for Button
 *
 * @namespace Button.Wrapper
 * @memberOf Button
  */

import styled from 'styled-components';


const Wrapper = styled.div`
  display: inline-block;
`;

export default Wrapper;
