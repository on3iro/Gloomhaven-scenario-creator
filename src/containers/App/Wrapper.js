/**
  * Styled Wrapper for the whole application
  *
  * @namespace App.Wrapper
  * @memberOf App
  */

import styled from 'styled-components';


const Wrapper = styled.div`
  height: 100%;
  min-height: 100%;
  display: flex;
  flex-direction: column;
`;

export default Wrapper;
