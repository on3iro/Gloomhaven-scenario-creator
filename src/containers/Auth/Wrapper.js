/**
  * Renders a styled div for the HomePage
  *
  * @namespace HomePage.Wrapper
  * @memberOf HomePage
  */

import styled from 'styled-components';


const Wrapper = styled.div`
  margin: 0 auto;
  width: 50%;
`;

export default Wrapper;
