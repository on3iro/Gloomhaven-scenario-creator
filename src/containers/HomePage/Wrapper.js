import styled from 'styled-components';


const Wrapper = styled.section`
  width: 50%;
  margin: 0 auto;
`;

export default Wrapper;
