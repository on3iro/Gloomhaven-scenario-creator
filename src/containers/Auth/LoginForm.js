/**
  * Renders a basic LoginForm
  *
  * @namespace HomePage.LoginForm
  * @memberOf HomePage
  */

import React, { PropTypes } from 'react';
import { Field, reduxForm } from 'redux-form';

import H1 from 'components/H1';
import Button from 'components/Button';

import RenderField from './RenderField';

const validate = values => {
  const errors = {};

  if(!values.email) {
    errors.email = 'Required';
  }else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
    errors.email = 'Invalid email address'
  }

  if(!values.password) {
    errors.password = 'Required';
  }

  return errors;
};

const LoginForm = props => {
  return (
    <form onSubmit={props.handleSubmit}>
      <H1>Login</H1>
      <Field name="email" type="text"
        component={RenderField} label="email"
      />
      <Field name="password" type="password"
        component={RenderField} label="password"
      />
      <Button submit >Login</Button>
    </form>
  );
}

LoginForm.propTypes = {
  // TODO
};

// Decorate the form component
export default reduxForm({
  form: 'login',
  validate
})(LoginForm);
