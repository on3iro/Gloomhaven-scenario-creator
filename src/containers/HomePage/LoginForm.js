/**
  * Renders a basic LoginForm
  *
  * @namespace HomePage.LoginForm
  * @memberOf HomePage
  */

import React, { Component } from 'react';

import H1 from 'components/H1';
import Button from 'components/Button';
import Input from 'components/Input';


class LoginForm extends Component {
  /**
    * Class component LoginForm
    *
    * @class LoginForm
    * @namespace LoginForm
    * @memberOf HomePage.LoginForm
    */

  constructor(props) {
    /**
      * Constructor of the LoginForm
      *
      * @constructs
      * @memberOf HomePage.LoginForm.LoginForm
      */

    super(props);

    this.state = {
      email: '',
      password: '',
    };
  }

  handleInputChange = (event) => {
    /**
      * Handles changes on input elements.
      * Sets the value of the respective input.
      *
      * @method handleInputChange
      * @memberOf HomePage.LoginForm.LoginForm
      * @namespace handleInputChange
      * @param {Object} event - Triggered event
      * @returns {undefined}
      */

    const target = event.target;
    const value = target.type === 'checkbox' ? target.checked : target.value;
    const name = target.name;

    this.setState({
      [name]: value,
    });

    return undefined;
  }

  render() {
    return (
      <form onSubmit={() => {}}>
        <H1>Login</H1>
        <Input
          name="email"
          type="text"
          placeholder="Email"
          value={this.state.email}
          onChange={this.handleInputChange}
        />
        <Input
          name="password"
          type="password"
          placeholder="Password"
          value={this.state.password}
          onChange={this.handleInputChange}
        />
        <Button submit>Login</Button>
      </form>
    );
  }
}

export default LoginForm;
