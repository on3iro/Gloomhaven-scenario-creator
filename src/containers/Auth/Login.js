import React, { PropTypes, Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import LoginForm from './LoginForm';
import Wrapper from './Wrapper';
import StyledLink from 'components/StyledLink';

import { loginSubmit } from './ducks/actions';


class Login extends Component {
  handleSubmit = values => {
    this.props.loginSubmit(values);
  }

  render() {
    return (
      <Wrapper>
        <LoginForm onSubmit={this.handleSubmit} />
        <StyledLink to="/register">Register</StyledLink>
      </Wrapper>
    );
  }
}

Login.propTypes = {
  // TODO
};

const mapDispatchToProps = dispatch => {
  return bindActionCreators({
    loginSubmit,
  }, dispatch);
};

export default connect(null, mapDispatchToProps)(Login);
