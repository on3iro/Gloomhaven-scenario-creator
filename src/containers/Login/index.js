/**
  * Login
  *
  * @namespace Login
  */

import React, { PropTypes, Component } from 'react';

import LoginForm from './LoginForm';
import Wrapper from './Wrapper';
import Button from 'components/Button';
import StyledLink from 'components/StyledLink';


class Login extends Component {
  handleSubmit = (values) => {
  }

  render() {
    return (
      <Wrapper>
        <LoginForm onSubmit={this.handleSubmit} />
        <StyledLink to="/register">Register</StyledLink>
      </Wrapper>
    );
  }
}

Login.propTypes = {
  // TODO
};


export default Login;
