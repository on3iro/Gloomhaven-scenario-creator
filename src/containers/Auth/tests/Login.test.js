import React from 'react';
import { shallow } from 'enzyme';

import StyledLink from 'components/StyledLink';

import Wrapper from '../Wrapper';
import { Login } from '../Login';
import LoginForm from '../LoginForm';


const renderComponent = newProps => {
  const props = {
    loginSubmit: jest.fn(),
    ...newProps,
  };

  const renderedWrapper = shallow(<Login {...props} />);

  return {
    props,
    renderedWrapper,
  };
};

describe('<Login />', () => {
  it('should render <Wrapper />', () => {
    const { renderedWrapper } = renderComponent();
    expect(renderedWrapper.find(Wrapper).length).toMatchSnapshot();
  });

  it('should render <LoginForm />', () => {
    const { renderedWrapper } = renderComponent();
    expect(renderedWrapper.find(LoginForm).length).toMatchSnapshot();
  });

  it('should render <StyledLink />', () => {
    const { renderedWrapper } = renderComponent();
    expect(renderedWrapper.find(StyledLink).length).toMatchSnapshot();
  });

  it('should call loginSubmit() on handleSubmit', () => {
    const { renderedWrapper, props } = renderComponent();

    renderedWrapper.instance().handleSubmit('test');

    expect(props.loginSubmit).toHaveBeenCalledWith('test');
  });
});
