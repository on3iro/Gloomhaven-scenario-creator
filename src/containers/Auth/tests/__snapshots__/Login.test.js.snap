exports[`<Login /> should render <LoginForm /> 1`] = `1`;

exports[`<Login /> should render <StyledLink /> 1`] = `1`;

exports[`<Login /> should render <Wrapper /> 1`] = `1`;
